### PR TITLE
feat(dashboard): theme assigning

### PR DESCRIPTION
**Pull request by minseolee**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description

I've made feature of theme assigning in dashboardContainer.

Only helper is made. Data is not connected to components.

스크린샷 2022-12-05 오후 8 05 57

### CODE REVIEW

File: src/services/dashboards/dashboard-detail/lib/themeHelper.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import type { WidgetThemeOption } from '@/services/dashboards/dashboard-detail/lib/type';
+
+const WIDGET_THEME = ['violet', 'blue', 'coral', 'yellow', 'gray', 'green', 'indigo', 'peacock'] as const;
+const WIDGET_THEME_LEN = WIDGET_THEME.length;
+
+// ['violet' | 'blue' | 'coral' ... | undefined]
+export type WidgetThemeAssignedList = Array<typeof WIDGET_THEME[number] | undefined>;
```
Comment (WANZARGEN) [inline src/services/dashboards/dashboard-detail/lib/themeHelper.ts:7]: suggestion: THEME constant is already exist(yuda made it) so you can reuse it :)

File: src/services/dashboards/dashboard-detail/lib/themeHelper.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+// function for circular array
+const circular = (n: number): number => (n % WIDGET_THEME_LEN + WIDGET_THEME_LEN) % WIDGET_THEME_LEN;
+
+export const widgetThemeAssigner = (widgetThemeOptionList: Array<WidgetThemeOption>): WidgetThemeAssignedList => {
```
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/lib/themeHelper.ts:12]: 🫶

File: src/services/dashboards/dashboard-detail/lib/type.ts [hunk 3]
```diff
@@ -2,3 +2,8 @@ import type { WIDGET_WIDTH_RANGE_LIST, WIDGET_WIDTH_FULL } from '@/services/dash
 
 // = 'SM' | 'MD' | 'LG' | 'XL' | 'FULL';
 export type WidgetSize = keyof typeof WIDGET_WIDTH_RANGE_LIST | typeof WIDGET_WIDTH_FULL;
+
+export interface WidgetThemeOption {
+    inherit: boolean;
+    inherit_count: number|undefined;
+}
```
Comment (WANZARGEN) [inline src/services/dashboards/dashboard-detail/lib/type.ts:9]: suggestion: I recommend to use `WidgetConfig` interface to reduce manage point :)

for example: 
```suggestion
export type WidgetThemeOption = WidgetConfig[theme]
```

File: src/services/dashboards/dashboard-detail/lib/themeHelper.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            widgetThemeAssignedList.push(WIDGET_THEME[circular(themeIndex)]);
+            themeIndex += 5;
+        }
+        if (!option.inherit) {
+            widgetThemeAssignedList.push(undefined);
+        }
+    });
+
+    return widgetThemeAssignedList;
+};
```
Comment (WANZARGEN) [inline src/services/dashboards/dashboard-detail/lib/themeHelper.ts:31]: praise: nice clean code :)

**Review by WANZARGEN — APPROVED** [review]
Nice clean code and nice test code!
I left some suggestions!

**Review by yuda110 — APPROVED** [review]
LGTM!

I think file names should be changed to `theme-helper.ts` and `width-helper.ts` :)

File: src/services/dashboards/dashboard-detail/lib/themeHelper.ts (hunk 1 above)
Comment (minseolee) [reply src/services/dashboards/dashboard-detail/lib/themeHelper.ts:7]: I'll apply it. Thanks !!!

**Pull request merged**